src/openzaak/utils/tasks.py

```diff
@@ -0,0 +1,765 @@
+import csv
+import functools
+import logging
+import shutil
+from contextlib import contextmanager
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+from time import monotonic
+from typing import Generator, Optional
+from uuid import uuid4
+
+from django.core.cache import cache
+from django.core.exceptions import ValidationError
+from django.db import Error as DatabaseError, IntegrityError, transaction
+from django.db.models.fields.files import FieldFile, FileDescriptor, FileField
+from django.utils import timezone
+from django.utils.functional import classproperty
+
+from rest_framework.test import APIRequestFactory
+from vng_api_common.constants import RelatieAarden
+from vng_api_common.utils import generate_unique_identification
+
+from openzaak import celery_app
+from openzaak.components.documenten.api.serializers import (
+    EnkelvoudigInformatieObjectSerializer,
+)
+from openzaak.components.documenten.models import (
+    EnkelvoudigInformatieObject,
+    EnkelvoudigInformatieObjectCanonical,
+)
+from openzaak.components.zaken.models.zaken import Zaak, ZaakInformatieObject
+from openzaak.utils.models import Import, ImportStatusChoices
+
+logger = logging.getLogger(__name__)
+
+
+def _get_csv_generator(filename: str) -> Generator[tuple[int, list], None, None]:
+    with open(filename, "r") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",", quotechar='"')
+
+        index = 1
+
+        for row in csv_reader:
+            yield index, row
+
+            index += 1
+
+
+def _get_total_count(filename: str, include_header: bool = False) -> int:
+    with open(filename, "r") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",", quotechar='"')
+        total = sum(1 for _ in csv_reader)
+
+        if include_header:
+            return total
+
+        return total - 1 if total > 0 else 0
+
+
+@dataclass
+class DocumentRow:
+    _identificatie: str
```

Comment:
Supporting both csv format and document serializer seems so bothersome and not worth it. Can we reuse `EnkelvoudigInformatieObjectSerializer` for it? We do it in Catalogi `import` command, and it adds up as an argument to use JSON format 

It will save us so much headache and will reduce operational risk significantly